### PR TITLE
cli: upgrade sub-command

### PR DESCRIPTION
**Pull request by dweomer**
Introducing `k3os upgrade`, replacing `k3os install` when `k3os.mode=local`.

`k3os upgrade --help`:

```
NAME:
   k3os upgrade - perform upgrades

USAGE:
   k3os upgrade [command options] [arguments...]

OPTIONS:
   --channel value  updates/releases uri (default: https://github.com/rancher/k3os/releases/latest) [$K3OS_UPGRADE_CHANNEL]
   --version value  release or tag (default: "latest") [$K3OS_UPGRADE_VERSION]
   --kernel         upgrade the kernel [$K3OS_UPGRADE_KERNEL]
   --rootfs         upgrade k3os+k3s [$K3OS_UPGRADE_ROOTFS]
   --remount        pre-upgrade remount? [$K3OS_UPGRADE_REMOUNT]
   --sync           post-upgrade sync? [$K3OS_UPGRADE_SYNC]
   --reboot         post-upgrade reboot? [$K3OS_UPGRADE_REBOOT]
```

### CODE REVIEW

**Comment by schmitch** [comment]
sorry to barge in, but doesn't this is a little bit naive to tie the implementation to github? I mean there is only a "github" channel, wouldn't it make more sense to make it a little bit more generic by having a well structured layout so that users can rsync stuff to mirror an upgrade server?

i.e. I deploy stuff to a company that uses sophos firewalls and it's basically impossible to allow stuff from github since github has a lot of stuff like downloads hosted on aws, which uses like too many ips behind der domain.

I would prefer a solution where I can have a "generic" channel which basically has a layout like that:

DOMAIN+PATH_PREFIX(can_be_empty)+/releases
|- /latest.json => {"version": "v1.0.0"}
|- /v1.0.0/k3os-kernel-version-amd64
|- /v1.0.0/k3os-initrd-amd64
|- /v1.0.0/k3os-kernel-amd64.squashfs
|- /v1.0.0/k3os-rootfs-amd64.tar.gz
|- /v1.0.0/k3os-vmlinuz-amd64
|- /v1.0.0/ sha256sum-amd64.txt
|- /v0.9.0/k3os-kernel-version-amd64
...

(I dropped ARM for consistensy)
So basically somebody can create a channel via rsync a "simple" layout to a server (s3/gcloud files) myabe even specifiying channel: URI (like https://k3os-releases.rancher.io/) and maybe channel: generic (Bonus points if Basic Auth or HMAC4 signature url/gcloud singature url would be supported)

well I think if my proposal is okay, I can even try to make an MR after your's is merged.

**Comment by dweomer** [comment]
@schmitch as you can see by this PR the go implementation has effectively supplanted the `k3os-upgrade-[kernel|rootfs]` scripts that were also bound to github via rest api calls. this PR does introduce the concept of a channel but it is currently effectively hard-coded to be github. this limitation should be temporary in that it will be lifted/resolved prior to the release of 1.0.

also, i am currently noodling over an alternative way to build and package k3os artifacts which will impact how upgrade/release channels work.

**Comment by dweomer** [comment]
closing in favor of work in #284
